nemo/collections/nlp/models/language_modeling/megatron_gpt_peft_models.py

```diff
@@ -268,7 +274,15 @@ def load_state_dict(self, state_dict, strict: bool = True):
 
     def setup_optimizer_param_groups(self):
         if self.first_stage_of_pipeline():
-            super().setup_optimizer_param_groups()
+            # super().setup_optimizer_param_groups()
+            self.freeze()  # Freeze the entire model
+            opt_params = []
+            for n, p in self.named_parameters():
+                if n in self.trainable_keys:
```

Comment:
Are trainable keys the full list of all the params of the adapters? Or just prefix of the adapter name?

Reply:
They are the list of full names of all adapter parameters which will be trained (i.e. weights will be updates). For backward compatibility reasons there are some adapter params which are not-trained but should still remain in the state_dict.